File: ruby/hamming/hamming.rb
class Hamming

  def self.compute (dna1, dna2)
    distance = 0

    raise ArgumentError if dna1.length != dna2.length

    idx = 0
    while (idx < dna1.length)
      if dna1[idx] != dna2[idx]
        distance += 1
      end
      idx += 1
    end

    return distance
  end
end


module BookKeeping
  VERSION = 3
end

File: javascript/hamming/hamming.js
// Create a module called Hamming
var Hamming = function() {}; 

// Add a method to calculate Hamming distance to the module
Hamming.prototype.compute = function(dna1, dna2){
	var distance = 0

	if (dna1.length !== dna2.length) {
		throw new Error("DNA strands must be of equal length."); 
	}

	var i = 0; 
	while (i < dna1.length) {
		if (dna1[i] !== dna2[i]) {
			distance++; 
		}
		i++; 
	}
	return distance
}

// Export the module
module.exports = Hamming; 


